kv/dist_sender.go

```diff
@@ -555,32 +548,52 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			var replicas replicaSlice
 			if err == nil {
 				replicas = newReplicaSlice(ds.gossip, desc)
-				rpcErr = ds.sendRPC(desc.RaftID, replicas, args, reply)
-				err = rpcErr
-			}
 
-			// If the RPC went through fine, an error (if any) is in the reply.
-			if err == nil {
-				err = reply.Header().GoError()
+				// Rearrange the replicas so that those replicas with long common
+				// prefix of attributes end up first. If there's no prefix, this is a
+				// no-op.
+				order := ds.optimizeReplicaOrder(replicas)
+
+				// Don't mess with replicas - we need the full list below for
+				// cache expiration considerations, but we may not want to send
+				// to all of them now.
+				useReplicas := replicas
```

Comment:
`useReplicas` and `replicas` are backed by the same array, which is potentially confusing when we modify `replicas` below (it's safe because we change `useReplicas` at the same time, but it's subtle enough to be broken by future changes). Add a comment about this. (we could also make useReplicas an int instead of a slice, and set it to either 1 or len(replicas), but I'm not sure it's worth it)